### PR TITLE
Fortran: Implement visit_ArraySection() and other fixes

### PR DESCRIPTION
**Pull request by Shaikh-Ubaid**

### CODE REVIEW

**Comment by Shaikh-Ubaid** [comment]
While working towards https://github.com/lfortran/lfortran/pull/2691, I implemented the ArraySection visitor in the fortran backend to help debug the issue. I polished the implementation, added other fixes for enabling the tests and submitted it as a PR here.

File: integration_tests/CMakeLists.txt [hunk 1]
```diff
@@ -996,8 +996,8 @@ RUN(NAME enum_01 LABELS gfortran llvm)
 RUN(NAME enum_02 LABELS gfortran llvm EXTRAFILES
          enum_02_module.f90)
 
-RUN(NAME array_section_01 LABELS gfortran llvm llvmStackArray)
-RUN(NAME array_section_02 LABELS gfortran llvm llvmStackArray)
+RUN(NAME array_section_01 LABELS gfortran llvm llvmStackArray fortran)
```
Comment (Shaikh-Ubaid) [inline integration_tests/CMakeLists.txt:999]: The `array_section_01` tests ArraySection visitor and variable dependency during their declaration/definition.

File: integration_tests/CMakeLists.txt [hunk 2]
```diff
@@ -996,8 +996,8 @@ RUN(NAME enum_01 LABELS gfortran llvm)
...
+RUN(NAME array_section_02 LABELS gfortran llvm llvmStackArray fortran)
```
Comment (Shaikh-Ubaid) [inline integration_tests/CMakeLists.txt:1000]: With respect to the commits in this PR, the `array_section_02` tests ArraySection, Function return variable declaration and FunctionCall.

**Comment by Shaikh-Ubaid** [comment]
The pointer type and Associate visitor are implemented, but not currently tested in this PR. I think they can be tested in a future PR.

**Review by certik — APPROVED** [review]

**Pull request merged**